### PR TITLE
search ui: follow up search result items improvements

### PR DESCRIPTION
**Pull request by limitedmage**
Follow up to #38785

- Removes hacky `overflow: hidden` properties on the search results page, instead relying on grid's `minmax` to prevent horizontal scrolling. This is necessary so sticky positioning works.
- Made the header of search results sticky so it stays at the top of the screen while scrolling
- Made the collapse button of the search results sticky so it stays at the bottom of the screen while scrolling
- Fixed the flash of content when expanding/collapsing search results when the core workflow improvements are enabled
  - Also fixed the text color of placeholder line numbers to match the color of loaded line numbers
- Fixed the repo result text color so it matches the other result types when the core workflow improvements are enabled

https://user-images.githubusercontent.com/206864/179296495-8276379e-664f-474b-b3e9-acade164610c.mp4

## Test plan

Test manually with feature flag on and off. No changes should occur in visual tests.

## App preview:

- [Web](https://sg-web-jp-searchfilesimprovements.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-hipcfhjnuu.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/components/ResultContainer.tsx [hunk 1]
```diff
@@ -234,10 +234,16 @@ export const ResultContainer: React.FunctionComponent<React.PropsWithChildren<Re
                     )}
                 </div>
                 <div className={classNames(coreWorkflowImprovementsEnabled && styles.collapsibleResults)}>
-                    {!expanded && collapsedChildren}
-                    {expanded && expandedChildren}
+                    <div>{expanded ? expandedChildren : collapsedChildren}</div>
```
Comment (limitedmage) [inline client/search-ui/src/components/ResultContainer.tsx:237]: The extra div wrapper is what prevents the flash of content when expanded/contracted. I think not having these wrapped and adding the button sibling causes React's change detection to get confused and it rerenders everything.

File: client/search-ui/src/components/CodeExcerpt.tsx [hunk 2]
```diff
@@ -215,7 +215,7 @@ export const CodeExcerpt: React.FunctionComponent<Props> = ({
                         <tbody>
                             {range(startLine, endLine).map(index => (
                                 <tr key={index}>
-                                    <td className="line">{index + 1}</td>
+                                    <td className="line" data-line={index + 1} />
```
Comment (limitedmage) [inline client/search-ui/src/components/CodeExcerpt.tsx:218]: This fixes the line numbers color issue as seen in this video where the line numbers are momentarily white

https://user-images.githubusercontent.com/206864/179296902-0f1ba97b-ad25-4282-b98e-553a332859de.mp4

File: client/search-ui/src/components/RepoSearchResult.tsx [hunk 3]
```diff
@@ -28,9 +29,17 @@ export const RepoSearchResult: React.FunctionComponent<RepoSearchResultProps> =
...
+    const [coreWorkflowImprovementsEnabled] = useCoreWorkflowImprovementsEnabled()
+
     const renderTitle = (): JSX.Element => (
         <div className={styles.title}>
-            <span className={classNames('test-search-result-label', styles.titleInner)}>
+            <span
+                className={classNames(
+                    'test-search-result-label',
+                    styles.titleInner,
+                    coreWorkflowImprovementsEnabled && styles.mutedRepoFileLink
```
Comment (limitedmage) [inline client/search-ui/src/components/RepoSearchResult.tsx:40]: Same code as `CommitSearchResult` and `FileSearchResult` to make the headers consistent.

**Review by fkling — APPROVED** [review]
The sticky header/foot looks slick! The only thing I noticed is that if you hide the additional matches it kind of stays where you are. I think I would expect it to scroll the search result that was just "closed" into view again. Does that make sense? Not sure if that's possible to do (and of course it doesn't have to be part of this PR), just a thought.

**Comment by limitedmage** [comment]
> I think I would expect it to scroll the search result that was just "closed" into view again. Does that make sense? Not sure if that's possible to do (and of course it doesn't have to be part of this PR), just a thought.

I'm trying to get this done but decided to split it into a separate PR. There are issues with this, mostly due to Safari still not fully supporting `scrollIntoView`. I'm playing around with it more.

**Pull request merged**